trin-core/src/portalnet/discovery.rs

```diff
@@ -285,3 +304,86 @@ impl Discovery {
         Ok(response)
     }
 }
+
+pub struct Discv5UdpSocket {
+    // `RwLock` for interior mutability.
```

Comment:
Is this `tokio::sync::Mutex` instead of `RwLock`?

Reply:
good catch! it was a `tokio::sync::RwLock` initially. I will update the comment.